Changes for 0.1.4

ulupo opened the pull request:
#### Reference Issues/PRs
(#97), (#120), (#121), (#122), (#123), (#125), (#127), (#128), (#153), (#158), (#170)

#### What does this implement/fix? Explain your changes.
Miscellaneous set of changes to prepare for the 0.1.4 release. The following is meant to guide the reviewer through the commit list but is not meant to be fully exhaustive.

- **Mapper-related**
  - Refactoring and renaming of plotting functions, with new and improved API and more efficient implementations (see f95b611 and f821687), fixing (#120), (#121), (#127), (#128), and (#125) for plotting
  - Removal of `Nerve` from `giotto.mapper.__init__.py` as we don't currently encourage its direct use
  - Changes in `ParallelClustering` (see ccdf704 and 208f9aa)
  - Small API change to `make_mapper_pipeline` (see 27a2c64)
  - Renaming of `set_mapper_params` to `set_params` so the `MapperPipeline` objects can be used in metaestimators or `GridSearchCV`s before knowing the nested structure (see 0be1927). Thanks @gtauzin for stressing the importance of doing this!
  - Better support for pandas dataframe input, partly addressing (#125)
  - Addition of Mapper to the sphinx documentation, adopting a novel subsection approach which could be used for the rest of the library. 

- **Docs-related**
  - Fixing (#97) by introducing a new module `giotto.utils._docs` with a class decorator `adapt_fit_transform_docs`
  - Linting, throughout (incomplete)
  - More consistency with scikit-learn documentation guidelines when declaring input types and shapes
  - Fixing (#123)

- **Other**
  - Adds CI and wheels for Python 3.8 (#153)
  - Temporarily fix (#170) by freezing the `pybind11` dependency to a specific point in time 
  - Adds `ipywidgets` to `requirements.txt`
  - Bumps `scikit-learn` dependency to 0.22.0
  - Fixing (#147), however non-optimally
  - Fixing (#122), but not yet in folders with test or in `plotting.py`
  - Updating license, project name, MAINTAINER variable, list of code authors, team members and maintainers (see 8220408, 81c4229, e526b21, 2185f07)
  - Commas in `KEYWORDS` variable in `setup.py`

Commit:
Remove Nerve from mapper __init__, reorder __all__

Commit:
Remove Nerve from mapper __init__, reorder __all__

Commit:
Document cluster.py, minor changes to ParallelClustering

- Write complete docstrings for ParallelClustering
- Remove `transform` in favour of `fit_transform`
- Remove seemingly unnecessary inheritance from ClusterMixin and TransformerMixin
- Turn `clusterer_` into hidden `_clusterer`
- Add `clusters_` attribute mirroring the `labels_` attribute of scikit-learn clusterers
- Improve docstrings for other classes in `cluster.py`
- Linting

Commit:
Document cluster.py, minor changes to ParallelClustering

- Write complete docstrings for ParallelClustering
- Remove `transform` in favour of `fit_transform`
- Remove seemingly unnecessary inheritance from ClusterMixin and TransformerMixin
- Turn `clusterer_` into hidden `_clusterer`
- Add `clusters_` attribute mirroring the `labels_` attribute of scikit-learn clusterers
- Improve docstrings for other classes in `cluster.py`
- Linting

Commit:
Update project name and list of authors in doc/conf.py

- Change project name to 'giotto-learn'
- Add @lewtun to list of authors

Commit:
Update project name and list of authors in doc/conf.py

- Change project name to 'giotto-learn'
- Add @lewtun to list of authors

Commit:
Edit MAINTAINER variable in setup.py

Update to new maintainers list

Commit:
Edit MAINTAINER variable in setup.py

Update to new maintainers list

Commit:
Add Mapper to sphinx docs

Commit:
Add Mapper to sphinx docs

Commit:
Add commas in KEYWORDS

Commit:
Add commas in KEYWORDS

Commit:
Add line before itemised lists

Commit:
Add line before itemised lists

Commit:
Small change in defaults and kwarg names in make_mapper_pipeline

- No longer using transformer instances as defaults for Mapper steps, all such defaults are now `None`.
- Change relatively uninformative `n_jobs_outer` into more explicit `parallel_clustering_n_jobs`.
- Add `parallel_clustering_prefer` argument which allows to set the type of parallelism to be used in ParallelClustering, if any.
- Replace **pipeline_kwargs with the two explicit kwargs from scikit-learn's `make_pipeline`, i.e. `memory` and `verbose`.
- Update validation steps inside function call accordingly.
- Change duck typing for `filter_func` to look for 'fit_transform' (as per TransformerMixin behaviour) instead of 'transform'.
- Update documentation accordingly.
- Update Mapper Quickstart notebook accordingly.
- Linting.

Commit:
Small change in defaults and kwarg names in make_mapper_pipeline

- No longer using transformer instances as defaults for Mapper steps, all such defaults are now `None`.
- Change relatively uninformative `n_jobs_outer` into more explicit `parallel_clustering_n_jobs`.
- Add `parallel_clustering_prefer` argument which allows to set the type of parallelism to be used in ParallelClustering, if any.
- Replace **pipeline_kwargs with the two explicit kwargs from scikit-learn's `make_pipeline`, i.e. `memory` and `verbose`.
- Update validation steps inside function call accordingly.
- Change duck typing for `filter_func` to look for 'fit_transform' (as per TransformerMixin behaviour) instead of 'transform'.
- Update documentation accordingly.
- Update Mapper Quickstart notebook accordingly.
- Linting.

Commit:
Place make_mapper_pipeline before MapperPipeline in index.rst

Commit:
Place make_mapper_pipeline before MapperPipeline in index.rst

Commit:
Globally replace "ndarray, shape" with "ndarray of shape"

This is consistent with scikit-learn's guidelines: https://scikit-learn.org/stable/developers/contributing.html#documentation

Commit:
Globally replace "ndarray, shape" with "ndarray of shape"

This is consistent with scikit-learn's guidelines: https://scikit-learn.org/stable/developers/contributing.html#documentation

Commit:
Fix (#123)

Globally replace "This method is there" with "This method is here"

Commit:
Fix (#123)

Globally replace "This method is there" with "This method is here"

Commit:
Minor rewording of `_version.py` docstring

Commit:
Minor rewording of `_version.py` docstring

Commit:
Add full stop

Commit:
Add full stop

Commit:
Add giotto.utils._docs.py module to fix (#97)

Implements a class decorator `adapt_fit_transform_docs` to generate the `fit_transform` docstring based on the `fit` and `transform` docstrings. Will fix (#97) when the decorator is applied to each class in giotto for which an adapted docstring is needed.

Commit:
Add giotto.utils._docs.py module to fix (#97)

Implements a class decorator `adapt_fit_transform_docs` to generate the `fit_transform` docstring based on the `fit` and `transform` docstrings. Will fix (#97) when the decorator is applied to each class in giotto for which an adapted docstring is needed.

Commit:
Decorate classes with adapt_fit_transform_docs (#97)

- Import and apply class decorator introduced by b9bb9eb 
- Optimize imports
- Linting

Commit:
Decorate classes with adapt_fit_transform_docs (#97)

- Import and apply class decorator introduced by b9bb9eb 
- Optimize imports
- Linting

Commit:
Decorate classes with adapt_fit_transform_docs (#97)

- Import and apply class decorator introduced by b9bb9eb 
- Optimize imports
- Linting

Commit:
Decorate classes with adapt_fit_transform_docs (#97)

- Import and apply class decorator introduced by b9bb9eb 
- Optimize imports
- Linting

Commit:
Decorate classes with adapt_fit_transform_docs (#97)

- Import and apply class decorator introduced by b9bb9eb 
- Optimize imports
- Linting

Commit:
Decorate classes with adapt_fit_transform_docs (#97)

- Import and apply class decorator introduced by b9bb9eb 
- Optimize imports
- Linting

Commit:
Decorate classes with adapt_fit_transform_docs (#97)

- Import and apply class decorator introduced by b9bb9eb 
- Optimize imports
- Linting

Commit:
Decorate classes with adapt_fit_transform_docs (#97)

- Import and apply class decorator introduced by b9bb9eb 
- Optimize imports
- Linting

Commit:
Decorate classes with adapt_fit_transform_docs (#97)

- Import and apply class decorator introduced by b9bb9eb 
- Optimize imports
- Linting

Commit:
Decorate classes with adapt_fit_transform_docs (#97)

- Import and apply class decorator introduced by b9bb9eb 
- Optimize imports
- Linting

Commit:
Decorate classes with adapt_fit_transform_docs (#97)

- Import and apply class decorator introduced by b9bb9eb 
- Optimize imports
- Linting

Commit:
Decorate classes with adapt_fit_transform_docs (#97)

- Import and apply class decorator introduced by b9bb9eb 
- Optimize imports
- Linting

Commit:
Decorate classes with adapt_fit_transform_docs (#97)

- Import and apply class decorator introduced by b9bb9eb 
- Optimize imports
- Linting

Commit:
Decorate classes with adapt_fit_transform_docs (#97)

- Import and apply class decorator introduced by b9bb9eb 
- Optimize imports
- Linting

Commit:
Decorate classes with adapt_fit_transform_docs (#97)

- Import and apply class decorator introduced by b9bb9eb 
- Optimize imports
- Linting

Commit:
Decorate classes with adapt_fit_transform_docs (#97)

- Import and apply class decorator introduced by b9bb9eb 
- Optimize imports
- Linting

Commit:
Add module docstring in nerve.py and TODO docs in Nerve

Commit:
Add module docstring in nerve.py and TODO docs in Nerve

Commit:
Decorate classes with adapt_fit_transform_docs (#97)

- Import and apply class decorator introduced by b9bb9eb 
- Optimize imports
- Linting

Commit:
Decorate classes with adapt_fit_transform_docs (#97)

- Import and apply class decorator introduced by b9bb9eb 
- Optimize imports
- Linting

Commit:
Decorate classes with adapt_fit_transform_docs (#97)

- Import and apply class decorator introduced by b9bb9eb 
- Optimize imports
- Linting

Commit:
Decorate classes with adapt_fit_transform_docs (#97)

- Import and apply class decorator introduced by b9bb9eb 
- Optimize imports
- Linting

Commit:
Decorate classes with adapt_fit_transform_docs (#97)

- Import and apply class decorator introduced by b9bb9eb 
- Optimize imports
- Replace :obj: with :func: in docs
- Linting
- Add TODO comment on mutable defaults

Commit:
Decorate classes with adapt_fit_transform_docs (#97)

- Import and apply class decorator introduced by b9bb9eb 
- Optimize imports
- Replace :obj: with :func: in docs
- Linting
- Add TODO comment on mutable defaults

Commit:
Decorate classes with adapt_fit_transform_docs (#97)

- Import and apply class decorator introduced by b9bb9eb
- Replace :obj: with :func: in docs
- Optimize imports
- Linting

Commit:
Decorate classes with adapt_fit_transform_docs (#97)

- Import and apply class decorator introduced by b9bb9eb
- Replace :obj: with :func: in docs
- Optimize imports
- Linting

Commit:
Decorate classes with adapt_fit_transform_docs (#97)

- Import and apply class decorator introduced by b9bb9eb
- Replace :obj: with :func: or :mod: in docs
- Optimize imports
- Linting
- Add TODO comment on mutable defaults

Commit:
Decorate classes with adapt_fit_transform_docs (#97)

- Import and apply class decorator introduced by b9bb9eb
- Replace :obj: with :func: or :mod: in docs
- Optimize imports
- Linting
- Add TODO comment on mutable defaults

Commit:
Decorate with adapt_fit_transform_docs (#97), relative imports (#122)

- Import and apply class decorator introduced by b9bb9eb
- Enforce relative imports as per (#122)
- Replace :obj: with :func: in docs
- Optimise imports
- Linting

Commit:
Decorate with adapt_fit_transform_docs (#97), relative imports (#122)

- Import and apply class decorator introduced by b9bb9eb
- Enforce relative imports as per (#122)
- Replace :obj: with :func: in docs
- Optimise imports
- Linting

Commit:
Decorate with adapt_fit_transform_docs (#97), relative imports (#122)

- Import and apply class decorator introduced by b9bb9eb
- Enforce relative imports as per (#122)
- Remove unnecessary _is_fitted attribute creation
- Optimise imports
- Linting

Commit:
Decorate with adapt_fit_transform_docs (#97), relative imports (#122)

- Import and apply class decorator introduced by b9bb9eb
- Enforce relative imports as per (#122)
- Remove unnecessary _is_fitted attribute creation
- Optimise imports
- Linting

Commit:
Refactor filter.py docs, decorate with adapt_fit_transform_docs (#97)

- Rewrite documentation for classes in filter.py
- Import and apply class decorator introduced by b9bb9eb
- Add TODO comment on having a stateful option for Eccentricity
- Add TODO comment on extending Projection functionality to pandas dataframes (#125)
- Linting

Commit:
Refactor filter.py docs, decorate with adapt_fit_transform_docs (#97)

- Rewrite documentation for classes in filter.py
- Import and apply class decorator introduced by b9bb9eb
- Add TODO comment on having a stateful option for Eccentricity
- Add TODO comment on extending Projection functionality to pandas dataframes (#125)
- Linting

Commit:
Rename is_fitted_ to _is_fitted

Commit:
Rename is_fitted_ to _is_fitted

Commit:
Remove Nerve import from `giotto.mapper.__init__.py`

Not done in c316dc3

Commit:
Remove Nerve import from `giotto.mapper.__init__.py`

Not done in c316dc3

Commit:
Add ipywidgets to requirements.txt

Azure pipelines triggered by (#137) failed. Likely cause: Azure pipelines had not been previously exposed to the visualization module (due also to the lack of tests for it), thereby not revealing the dependency on ipywidgets. c316dc3 triggered failure by adding visualization functions to the mapper module imports.

Commit:
Add ipywidgets to requirements.txt

Azure pipelines triggered by (#137) failed. Likely cause: Azure pipelines had not been previously exposed to the visualization module (due also to the lack of tests for it), thereby not revealing the dependency on ipywidgets. c316dc3 triggered failure by adding visualization functions to the mapper module imports.

Commit:
Improve documentation in mapper pipeline module

Commit:
Improve documentation in mapper pipeline module

Commit:
Reinstate `transform` in ParallelClustering as not yet implemented

The removal of `transform` by ccdf704 triggers errors when MapperPipeline objects are instantiated, because scikit-learn requires that `transform` methods be implemented by all intermediate transformers of a Pipeline object.

Reinstating it as a not yet implemented method has the following consequences:
- MapperPipeline objects can be instantiated with no errors
- The behaviour upon calling `fit_transform` is the correct one as the implemented `fit_transform` method is used.
- calling `fit` on a MapperPipeline object and then `transform` will fail as a NotImplementedError is raised. 

The last point is likely a good thing as we avoid suggesting to the user that assignments of new points to clusters seen in `fit` are available.

Commit:
Reinstate `transform` in ParallelClustering as not yet implemented

The removal of `transform` by ccdf704 triggers errors when MapperPipeline objects are instantiated, because scikit-learn requires that `transform` methods be implemented by all intermediate transformers of a Pipeline object.

Reinstating it as a not yet implemented method has the following consequences:
- MapperPipeline objects can be instantiated with no errors
- The behaviour upon calling `fit_transform` is the correct one as the implemented `fit_transform` method is used.
- calling `fit` on a MapperPipeline object and then `transform` will fail as a NotImplementedError is raised. 

The last point is likely a good thing as we avoid suggesting to the user that assignments of new points to clusters seen in `fit` are available.

Commit:
Change license comments throughout

Globally replace Apache 2.0 with GNU AGPLv3

Commit:
Change license comments throughout

Globally replace Apache 2.0 with GNU AGPLv3

Commit:
Remove newline in docstring

Commit:
Remove newline in docstring

Commit:
Remove unused imports in `giotto.pipeline.py`

Commit:
Remove unused imports in `giotto.pipeline.py`

Commit:
Rename parameters in ParallelClustering

Rename `n_jobs_outer` with `parallel_clustering_n_jobs` and `prefer` with `parallel_clustering_prefer` so that `set_mapper_param` can work after the changes in 
27a2c64

Commit:
Rename parameters in ParallelClustering

Rename `n_jobs_outer` with `parallel_clustering_n_jobs` and `prefer` with `parallel_clustering_prefer` so that `set_mapper_param` can work after the changes in 
27a2c64

Commit:
Reflect changes from  61c607a  in pipeline.py, docs improvements

- Make `make_mapper_pipeline` compatible with changes in 61c607a, add 'parallel_clustering_n_jobs' and 'parallel_clustering_prefer' to clust_params variable to make `set_mapper_param` work
- Extensively rewrite Examples section of `make_mapper_pipeline` so it is fairly exhaustive

Commit:
Reflect changes from  61c607a  in pipeline.py, docs improvements

- Make `make_mapper_pipeline` compatible with changes in 61c607a, add 'parallel_clustering_n_jobs' and 'parallel_clustering_prefer' to clust_params variable to make `set_mapper_param` work
- Extensively rewrite Examples section of `make_mapper_pipeline` so it is fairly exhaustive

lewtun commented on examples/plotting.py:
```diff
@@ -15,7 +15,7 @@ def plot_point_cloud(point_cloud, dimension=None):
 
```
Since we now include Plotly as a requirement with the library, would it make sense to (eventually) migrate these plotting functions to the core library? Personally, I cannot think of many situations where I would study persistent homology without visualising the diagrams. If not part of this PR, we could just add a TODO

ulupo replied:
Absolutely, well spotted!

ulupo replied:
Let's add a TODO, perhaps in a new project dedicated to 0.1.4?

lewtun replied:
Yes, sounds definitely worthy of a separate issue - I'll add one now

lewtun commented on giotto/base.py:
```diff
@@ -1,6 +1,6 @@
 """Implements a TransformerResamplerMixin for transformers that have a resample
 method."""
-# License: Apache 2.0
+# License: GNU AGPLv3
```
Can we get rid of these license comments? They're a) annoying to maintain and b) should be covered by the LICENSE file itself.

ulupo replied:
I have no strong feelings about this. Let's ask @matteocao once he is back.

lewtun replied:
Issue created

lewtun commented on giotto/diagrams/_metrics.py:
```diff
@@ -1,13 +1,14 @@
-# License: Apache 2.0
+# License: GNU AGPLv3
```
Ditto as above

lewtun commented on giotto/diagrams/_utils.py:
```diff
@@ -1,5 +1,5 @@
 """Utility functions for diagrams."""
-# License: Apache 2.0
+# License: GNU AGPLv3
```
Ditto as above. I'll stop repeating the same comment from now on.

lewtun commented on giotto/diagrams/distance.py:
```diff
@@ -176,8 +180,9 @@ def transform(self, X, y=None):
         -------
-        Xt : ndarray, shape (n_samples_fit, n_samples, n_homology_dimensions) \
-             if `order` is ``None``, else (n_samples_fit, n_samples)
+        Xt : ndarray of shape (n_samples_fit, n_samples, \
```
Does this line split render nicely in the docs? Otherwise it would be better to introduce the line separation earlier to get the tuple in one line.

ulupo replied:
It does at my end.

lewtun commented on giotto/diagrams/preprocessing.py:
```diff
@@ -1,24 +1,27 @@
 
-    Useful when downstream tasks require using topological features
-    all at once -- and not separated between different homology dimensions.
+    Useful when downstream tasks require using topological features all at
```
"using" -> "use of"

ulupo replied:
I'd have thought "the use of", in that case?

lewtun replied:
ah yes you're right!

ulupo replied:
Done.

lewtun commented on giotto/diagrams/preprocessing.py:
```diff
@@ -245,18 +250,18 @@ def transform(self, X, y=None):
 
-    def inverse_transform(self, X, copy=None):
+    def inverse_transform(self, X):
         """Scale back the data to the original representation. Multiplies
```
Would "Rescale the data" be clearer than "Scale back"?

ulupo replied:
Not sure. I find "rescaling" to be an ambiguous word, as in my experience it is sometimes used as a synonim of "scaling".

lewtun replied:
Ok, let's leave it as is

lewtun commented on giotto/diagrams/preprocessing.py:
```diff
@@ -267,6 +272,7 @@ def inverse_transform(self, X, copy=None):
 
 
+@adapt_fit_transform_docs
 class Filtering(BaseEstimator, TransformerMixin):
```
Maybe not worth it, but a better name for this class could be "FilterDiagrams" or "FilterPersistenceDiagrams". The use of the "ing" suffix is a bit weird, especially since we do not use e.g. "Scaling"

ulupo replied:
Agree about the weirdness of the gerund, but I'd also avoid putting "Diagrams" in there for brevity and also because it is already a class in giotto.diagrams

lewtun replied:
Fine with me, it was just a suggestion

lewtun commented on giotto/graphs/geodesic_distance.py:
```diff
@@ -63,13 +67,15 @@ def _geodesic_distance(self, X):
+        X : ndarray of shape (n_samples,) or \
+            (n_samples, n_vertices, n_vertices)
             Input data, i.e. a collection of adjacency matrices of graphs.
+            Each adjacency matrix may be a dense or a sparse array.
```
You could remove "a" from "or a" if desired. That is "dense or sparse array."

ulupo replied:
Indeed. Done.

Commit:
Remove f-string formatting for compatibility with Python 3.5

Replace f-formatting with str.format for backward compatibility

Commit:
Remove f-string formatting for compatibility with Python 3.5

Replace f-formatting with str.format for backward compatibility

lewtun commented on giotto/base.py:
```diff
@@ -1,6 +1,6 @@
 """Implements a TransformerResamplerMixin for transformers that have a resample
```
Are these comments really needed? Ideally the code itself would guide the reader and removing these comments would save on needing to maintain them (e.g. if the module is extended).

ulupo replied:
It's required by flake8, this I recall is the main reason we have these docstrings.

lewtun commented on giotto/mapper/cover.py:
```diff
@@ -11,8 +14,10 @@
 
+@adapt_fit_transform_docs
 class OneDimensionalCover(BaseEstimator, TransformerMixin):
     """Covers of one-dimensional data coming from open overlapping intervals.
```
Should it be "Cover" instead of "Covers" since we only get one cover?

ulupo replied:
Well spotted!

ulupo replied:
Done.

lewtun commented on giotto/mapper/cover.py:
```diff
@@ -316,6 +321,7 @@ def _cover_limits(min_val, max_val, n_intervals, overlap_frac):
 
+@adapt_fit_transform_docs
 class CubicalCover(BaseEstimator, TransformerMixin):
     """Covers of multi-dimensional data coming from overlapping hypercubes
```
Ditto as above regarding "Covers" vs "Cover"

ulupo replied:
Done.

lewtun commented on giotto/mapper/filter.py:
```diff
@@ -6,23 +9,35 @@
+    Let `D` be a square matrix representing distances between points in a
+    point cloud, or directly defining an abstract metric (or metric-like)
+    space. The eccentricity of point `i` in the point cloud or abstract
+    metric space is the `p` -norm (for some `p`) of row `i` in `D`.
```
Remove space between `p` and "-norm"

ulupo replied:
Done.

lewtun commented on giotto/mapper/pipeline.py:
```diff
@@ -114,28 +139,52 @@ def make_mapper_pipeline(scaler=MinMaxScaler(),
+
+    min_intersection : int, optional, default: ``1``
         Minimum size of the intersection between clusters required for
         creating an edge in the final Mapper graph.
```
could remove "final"

ulupo replied:
Done.

lewtun commented on giotto/utils/_docs.py:
```diff
@@ -0,0 +1,134 @@
+"""Utilities for docstring building."""
+# License : Apache 2.0
```
If we really need to include the GPL license on every file, then we should probably include it here too

ulupo replied:
Well spotted, I missed it in my refactoring due to the space before the colon!

ulupo replied:
Done.

lewtun reviewed (commented):
Overall the PR looks really good - the docstring decorator is ace!

Commit:
Docstring and comment fixes following review of (#137)

Commit:
Docstring and comment fixes following review of (#137)

Commit:
Wording change following review of (#137)

Commit:
Wording change following review of (#137)

lewtun reviewed (approved):
I would not merge until we have found a good solution for python_igraph on Windows.

Commit:
Add validate kwarg to `identity` (#125)

In this way, when `scaler=None` in `make_mapper_pipeline` we can have downstream filter functions acting on pandas dataframes directly.

Commit:
Add validate kwarg to `identity` (#125)

In this way, when `scaler=None` in `make_mapper_pipeline` we can have downstream filter functions acting on pandas dataframes directly.

Commit:
Better pandas dataframe support for `Projection` (#125)

- Replace `column_indices` init parameter with `columns`
- `columns` can be a list of pandas column names
- Duck typing determines if we are likely seeing a pandas dataframe in `transform`

Commit:
Better pandas dataframe support for `Projection` (#125)

- Replace `column_indices` init parameter with `columns`
- `columns` can be a list of pandas column names
- Duck typing determines if we are likely seeing a pandas dataframe in `transform`

Commit:
Grammar change

Commit:
Grammar change

Commit:
Add __init__ docstring for mapper module and mapper.pipeline submodule

Plus some linting in other __init__ docstrings

Commit:
Add __init__ docstring for mapper module and mapper.pipeline submodule

Plus some linting in other __init__ docstrings

Commit:
Add __init__ docstring to mapper.visualization

Commit:
Add __init__ docstring to mapper.visualization

Commit:
Replace `column_indices` with `columns` in Mapper Quickstart notebook

- For consistency with the changes to `Projection` made in 4ab49ab
- Also remove output for lighter diffs

Commit:
Replace `column_indices` with `columns` in Mapper Quickstart notebook

- For consistency with the changes to `Projection` made in 4ab49ab
- Also remove output for lighter diffs

Commit:
Linting in visualization.py

Commit:
Linting in visualization.py

Commit:
Refactoring of create_network_2d to fix (#121), (#125), (#127), (#128)

- A suggestion of a template for fixing all the above issues in the static and 2d case.
- Some necessary changes also performed in `utils.visualization.py`
- Mapper Quickstart notebook updated to show new behaviour with a pandas dataframe example

NOTE IMPORTANT: this has broken create_interactive_network, which will need to be refactored separately if these changes are accepted. Also, no change has been made to create_network_3d, pending approval for 2d.

lewtun committed:
Merge branch 'master' into 0.1.4

lewtun committed:
Merge remote-tracking branch 'origin/0.1.4' into 0.1.4

lewtun committed:
Merge branch '0.1.4' into fix_mapper_visualization

lewtun committed:
Rename node_color argument to node_color_statistic

lewtun committed:
Refactor name of is_node_colors_ndadday to is_node_colors_ndarray

lewtun committed:
Extend functionality from 2D visualisation to 3D

- Does not include the hoverlabel

lewtun committed:
Set hoverlabel background color for 3D visualisation

lewtun committed:
Fix linting in visualisation

lewtun committed:
Extend interactive visualisation to include new API

lewtun committed:
Fix linting

lewtun committed:
Add "Color by" annotation to 3D visualisation

lewtun committed:
Move viridis colorscale to top of dropdown list

Since the default colorscale is viridis, it makes sense to show the user this value at the top of the dropdown

lewtun committed:
Refactor 2D and 3D networks into single function

lewtun committed:
Merge plot_options from 2D and 3D networks and remove colorscale dropdown

lewtun committed:
Merge plot and layout options from 2D and 3D plots

lewtun committed:
Change sliders for filter params to bounded boxes

lewtun committed:
Fix layout options for 2D and 3D plots

lewtun committed:
Add docstring for dim argument

lewtun committed:
Refactor case handling for 2D vs 3D plotting into single if statement

lewtun committed:
Refactor 3D if statement to single clause

lewtun committed:
Remove 2D and 3D plotting functions

lewtun committed:
Remove canonical node ID from hover text

lewtun committed:
Update init for static function

lewtun committed:
Add 3D functionality to interactive plot

lewtun committed:
Clone pipeline in static plotting

lewtun committed:
Update quickstart notebook for new API changes

lewtun committed:
Modularise case checking to helper functions

lewtun committed:
Update quickstart notebook for new API

lewtun committed:
Fix bug with layout dim refactoring

lewtun committed:
Remove unused helper function for color scales

lewtun committed:
Sort imports in visualisation.py

Commit:
[DOC] wording change in VietorisRipsPersistence.transform

Commit:
Update CODE_AUTHORS, GOVERNANCE, setup.py

- Update with current list of maintainers, authors, and Giotto project team
- Add Mapper to keywords

Commit:
Update license in setup.py

Commit:
Remove space for consistency

Commit:
Modify CLASSIFIERS in setup.py

C -> C++

Commit:
Merge branch 'master' into 0.1.4

Commit:
Fix check_is_fitted for scikit-learn 0.22.1 (#119)

- From v0.22.1, attributes with a leading underscore are no longer sufficient for check_is_fitted to pass. Reintroduce explicit attribute names in cases where attributes with trailing underscores (and no leading dunderscore) are available. NOTE: this reintroduces warnings in scikit-learn v0.22.0.
- Linting

lewtun committed:
Add TODO comment on JIT computation for color by columns

Commit:
Fix (#147) (temporary)

- sklearn.cluster._hierarchical was renamed to sklearn.cluster._agglomerative in scikit-learn v0.22.1 (#119). (#147) is fixed in a temporary way by exception handling in the import statement.
- Reinstate scikit-learn >= 0.22.0 in requirements.txt

NOTE: This fix may be OK for 0.1.4 but cannot be regarded as long-term. In general it is bad practice to rely explicitly on hidden external modules. A longer-term fix is required for future releases (>= 0.1.5).

Commit:
Change ModuleNotFoundError to ImportError for Python 3.5 compatibility

The ModuleNotFoundError exception was introduced in 3.6 and is a subclass of ImportError

Commit:
Updade README and requirements for consistency

- Partition requirements list in README into what is needed for the latest stable release (now 0.1.3) and what is needed for dev/nightly
- Mention about nightly and explain what it is about
- Reorder requirements

Commit:
Update README and requirements for consistency

- Partition requirements list in README into what is needed for the latest stable release (now 0.1.3) and what is needed for dev/nightly
- Mention about nightly and explain what it is about
- Reorder requirements

Commit:
Merge remote-tracking branch 'origin/0.1.4' into 0.1.4

lewtun committed:
Change name of hoverlabel text variables

lewtun committed:
Refactor name of _is_array_or_dataframe() utility function

lewtun committed:
Fix docstring to reflect change from node_colors to node_color_statistic

lewtun committed:
Fix name of attribute names in docstrings and use US english for color

Commit:
Bump plotly version to latest distributed by conda

lewtun committed:
Add clone_pipeline bool and remove get_figure() utility function

lewtun committed:
Migrate examples docstring to issue #152

lewtun committed:
Refactor setting of cover parameters to be generic for ints and floats

lewtun committed:
Add string-based widget for cover

lewtun committed:
Change color_by_column default value to False

lewtun committed:
Tidy up color_variable explanation in docstrings

lewtun committed:
Update quickstart notebook to match API changes

lewtun committed:
Remove unused get_figure() helper function

Commit:
Merge branch 'master' of https://github.com/giotto-ai/giotto-learn into 0.1.4

lewtun committed:
Remove submit button by setting continuous_update False

lewtun committed:
Clean up commented out code from removal of submit button

lewtun committed:
Add comment about cloning pipeline

lewtun committed:
Refactor update_figure() to accept node and edge traces

Saves memory by avoiding need to create a new plotly Figure object with each update

Commit:
Merge branch 'master' of https://github.com/giotto-ai/giotto-learn into 0.1.4

Commit:
Replace remaining mentions of Apache 2.0 license

lewtun committed:
Clean up function signatures and variable names for node and edge traces

lewtun committed:
Tweak whitespace in comment

lewtun committed:
Refactor helper function to reflect changes to return

lewtun committed:
Update quickstart notebook to reflect changes

lewtun committed:
Fix docstring to remove attr

lewtun committed:
Fix docstring for color by columns dropdown default value

ulupo committed:
Merge pull request #1 from ulupo/fix_mapper_visualization

* Refactoring of create_network_2d to fix (#121), (#125), (#127), (#128)

- A suggestion of a template for fixing all the above issues in the static and 2d case.
- Some necessary changes also performed in `utils.visualization.py`
- Mapper Quickstart notebook updated to show new behaviour with a pandas dataframe example

NOTE IMPORTANT: this has broken create_interactive_network, which will need to be refactored separately if these changes are accepted. Also, no change has been made to create_network_3d, pending approval for 2d.

* Rename node_color argument to node_color_statistic

* Refactor name of is_node_colors_ndadday to is_node_colors_ndarray

* Extend functionality from 2D visualisation to 3D

- Does not include the hoverlabel

* Set hoverlabel background color for 3D visualisation

* Fix linting in visualisation

* Extend interactive visualisation to include new API

* Fix linting

* Add "Color by" annotation to 3D visualisation

* Move viridis colorscale to top of dropdown list

Since the default colorscale is viridis, it makes sense to show the user this value at the top of the dropdown

* Refactor 2D and 3D networks into single function

* Merge plot_options from 2D and 3D networks and remove colorscale dropdown

* Merge plot and layout options from 2D and 3D plots

* Change sliders for filter params to bounded boxes

* Fix layout options for 2D and 3D plots

* Add docstring for dim argument

* Refactor case handling for 2D vs 3D plotting into single if statement

* Refactor 3D if statement to single clause

* Remove 2D and 3D plotting functions

* Remove canonical node ID from hover text

* Update init for static function

* Add 3D functionality to interactive plot

* Clone pipeline in static plotting

* Update quickstart notebook for new API changes

* Modularise case checking to helper functions

* Update quickstart notebook for new API

* Fix bug with layout dim refactoring

* Remove unused helper function for color scales

* Sort imports in visualisation.py

* Add TODO comment on JIT computation for color by columns

* Change name of hoverlabel text variables

* Refactor name of _is_array_or_dataframe() utility function

* Fix docstring to reflect change from node_colors to node_color_statistic

* Fix name of attribute names in docstrings and use US english for color

* Add clone_pipeline bool and remove get_figure() utility function

* Migrate examples docstring to issue #152

* Refactor setting of cover parameters to be generic for ints and floats

* Add string-based widget for cover

* Change color_by_column default value to False

* Tidy up color_variable explanation in docstrings

* Update quickstart notebook to match API changes

* Remove unused get_figure() helper function

* Remove submit button by setting continuous_update False

* Clean up commented out code from removal of submit button

* Add comment about cloning pipeline

* Refactor update_figure() to accept node and edge traces

Saves memory by avoiding need to create a new plotly Figure object with each update

* Clean up function signatures and variable names for node and edge traces

* Tweak whitespace in comment

* Refactor helper function to reflect changes to return

* Update quickstart notebook to reflect changes

* Fix docstring to remove attr

* Fix docstring for color by columns dropdown default value

Co-authored-by: Umberto Lupo <46537483+ulupo@users.noreply.github.com>

ulupo committed:
Formatting fix in README

Commit:
Add "GNU" to license name in README

ulupo committed:
Link to GitHub homepage of giotto.ai in README

ulupo committed:
Add Community section with Slack workspace link

lewtun committed:
Fix bug where hover text would not display for all nodes

Figure update also requires an update of the color by column buttons

Commit:
Fix typo

Commit:
Linting for PEP8 compliance

ulupo committed:
Merge pull request #3 from ulupo/fix_hovertext

Fix hovertext bug

Commit:
Fix cloud links to wheels for pycairo in azure pipelines

Commit:
Attempt to remove wget from windows parts of azure pipelines

Commit:
Add conditional to insert python-igraph requirement

- Add conditional in setup.py
- Remove pip install python-igraph from azure pipelines in non-windows platforms

Commit:
Remove plotly and matplotlib from azure pipelines

Made redundant by requirements now explicitly mentioning both dependencies

Commit:
Add python-igraph to requirements.txt, remove install step from azure

- python-igraph added in requirements.txt
- case handling detects if system is windows, in which case the dependency_links argument of `setup` is passed a list of two URLs for pycairo and python-igraph
- Remove installation step from remotely hosted wheels from azure pipelines as this step should already be handles by setup.py

Commit:
Remove garbled line from setup.py

Commit:
Remove python-igraph from requirements.txt

Do case handling for INSTALL_REQUIRES to handle Windows

Commit:
Remove dependency_links, use approach based on PEP8 direct reference

As described in https://stackoverflow.com/questions/12518499/pip-ignores-dependency-links-in-setup-py#comment94736800_13587734

Commit:
Add Python 3.8 to azure pipelines (#153)

lewtun committed:
Refactor mapper plotting functions to match sklearn naming convention

lewtun committed:
Refactor mapper quickstart notebook to reflect API name changes

lewtun committed:
Update index.rst to reflect new API

ulupo committed:
Merge pull request #4 from ulupo/refactor_plotting_names

Refactor plotting names

Commit:
add temporary fix for pybind11

This allow to build wheels after pybind11 change in commit 07e22593

Signed-off-by: julian <julian.burellaperez@heig-vd.ch>

Commit:
Add clustering_preprocessing kwarg (#149)

Add optional transformer kwarg to make_mapper_pipeline to allow advanced users to transform data so ListFeatureUnion returns the transformed data in the first entry. Feature follows request in (#149)

ulupo committed:
Merge pull request #6 from reds-heig/fix_windows_building

add temporary fix for pybind11

Commit:
Merge remote-tracking branch 'origin/0.1.4' into 0.1.4

ulupo committed:
Merge branch 'master' into 0.1.4

lewtun committed:
Reshape output from FunctionTransformer for compatibility with MapperPipeline

lewtun committed:
Add simple example of passing callables to mapper quickstart notebook

Commit:
Rename set_mapper_params to set_params

Change should allow for mapper pipelines to be used easily as parts of metaestimators and GridSearchCV objects

ulupo committed:
Merge pull request #7 from ulupo/reshape_for_callable_filters

Reshape for callable filters

Commit:
Remove unused BaseEstimator import

From mapper.pipeline

lewtun committed:
Refactor Mapper quickstart notebook to match API and provide extra info

lewtun committed:
Simplify naming of void notebook

H2 on the plane assumes too prior knowledge for beginner users

lewtun committed:
Add pandas example to mapper notebook

lewtun committed:
Remove output from notebooks

ulupo committed:
Merge pull request #8 from ulupo/refactor_notebooks

Refactor notebooks

Commit:
Standardise TODOs

Commit:
Add summary description and fix return values in plotting functions

Commit:
Add original Mapper paper to references

Commit:
Document Nerve class, apply adapt_fit_transform_docs, hide attributes

But Nerve needs to be refactored to remove fit and transform in favour of fit_transform. It is misleading to have a transform method which completely ignores X and only uses the X seen in fit. Nerve needs to be turned into a pure fit-transformer (implement only fit_transform), but this means the MapperPipeline output by make_mapper_pipeline needs to be augmented with a finall 'passthrough' step since otherwise the scikit-learn API would complain about the last step not implementing a fit method.

Commit:
Document Nerve class, apply adapt_fit_transform_docs, hide 

But Nerve needs to be refactored to remove fit and transform in favour of fit_transform. It is misleading to have a transform method which completely ignores X and only uses the X seen in fit. Nerve needs to be turned into a pure fit-transformer (implement only fit_transform), but this means the MapperPipeline output by make_mapper_pipeline needs to be augmented with a finall 'passthrough' step since otherwise the scikit-learn API would complain about the last step not implementing a fit method.

Commit:
Merge remote-tracking branch 'origin/0.1.4' into 0.1.4

Commit:
Remove transform method from Nerve, add fit_transform, unhide attributes

Attributes are only stored if fit is called.

Commit:
Add documentation for fit attributes in Nerve

Commit:
Add optional graph_step in make_mapper_pipeline

For advanced users who prefer to have the pipeline stopped at the clustering step.

Commit:
Document method_to_transform

- Also remove unnecessary check for whether transform already exists. Makes no difference as transform is overwritten anyway.

Commit:
func_from_callable_on_rows -> transformer_from_callable_on_rows

- Utility function now returns the FunctionTransformer directly for more convenient use
- Better handling of shapes in the non-numpy-shortcut case via a helper function
- Add documentation
- Add to mapper-level init
- Add to index.rst in doc
- Modify make_mapper_pipeline accordingly

Commit:
Remove unused imports

Commit:
Add missing space

Commit:
Rename notebooks to have consistent snake_case style

Commit:
Add trivial __init__ to Entropy for docs

Commit:
Minor docstring improvements

Commit:
Rename Mapper node metadata keys

- Rename 'interval_id' to 'pullback_set_label'
- Rename 'cluster_id' to 'partial_cluster_label'
- Linting

Commit:
Remove unnecessary explicit return statement
